server/data/data-api/src/main/java/org/apache/james/user/api/DelegationStore.java

```diff
@@ -0,0 +1,36 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.user.api;
+
+import org.apache.james.core.Username;
+import org.reactivestreams.Publisher;
+
+public interface DelegationStore {
+    /**
+     * @return Lists of the users authorized to impersonnate to baseUser.
+     */
+    Publisher<Username> authorizedUsers(Username baseUser);
+
+    Publisher<Void> clear(Username baseUser);
+
+    Publisher<Void> addAuthorizedUser(Username baseUser, Username userWithAccess);
+
+    Publisher<Void> removeAuthorizedUser(Username baseUser, Username userWithAccess);
```

Comment:
Maybe we could have a little java doc for other APIs?

Reply:
BTW should we have an API to check if a user is authorized to impersonate to baseUser?

Reply:
> BTW should we have an API to check if a user is authorized to impersonate to baseUser?

That's what the `Authorizator` API is doing ;-)

Reply:
> Maybe we could have a little java doc for other APIs?

Boiler plate. I don't like java doc. I added that one so that we do not mix the delegated user and the base user. It seemed uneeded for other methods. Even this javadoc addition is questionable IMO.